manifests/templates/cicd/build.yaml

```diff
@@ -0,0 +1,143 @@
+- apiVersion: cyclone.io/v1alpha1
+  kind: Stage
+  metadata:
+    name: build-template
+    labels:
+      cyclone.io/stage-template: "true"
+      cyclone.io/scene: cicd
+      cyclone.io/template-kind: build
+    annotations:
+      cyclone.io/description: >
+        Build source code
+  spec:
+    pod:
+      inputs:
+        arguments:
+        - name: image
+        - name: cmd
+        - name: output-artifact-path
+        resources:
+        - name: code
+          path: /workspace
+      outputs:
+        artifacts:
+        - name: bin
+          path: "{{ output-artifact-path }}"
+      spec:
+        containers:
+        - image: "{{ image }}"
+          command:
+          - /bin/sh
+          - -c
+          - "{{ cmd }}"
+- apiVersion: cyclone.io/v1alpha1
+  kind: Stage
+  metadata:
+    name: golang-build-template
+    labels:
+      cyclone.io/stage-template: "true"
+      cyclone.io/scene: cicd
+      cyclone.io/template-kind: build
+    annotations:
+      cyclone.io/description: >
+        Build Golang source code
+  spec:
+    pod:
+      inputs:
+        arguments:
+        - name: image
+          value: golang:1.10-alpine3.8
+        - name: cmd
+          value: |
+            # Replace <main-entrypoint-path> with the main package path, for example 'cyclone/cmd/workflow/controller'.
+            go build -o /go/bin/app <main-entrypoint-path>
+        - name: output-artifact-path
+          value: /go/bin/app
+        resources:
+        - name: code
+          path: /go/src
+      outputs:
+        artifacts:
+        - name: bin
+          path: "{{ output-artifact-path }}"
+      spec:
+        containers:
+        - image: "{{ image }}"
+          command:
+          - /bin/sh
+          - -c
+          - "{{ cmd }}"
+- apiVersion: cyclone.io/v1alpha1
+  kind: Stage
+  metadata:
+    name: maven-build-template
+    labels:
+      cyclone.io/stage-template: "true"
+      cyclone.io/scene: cicd
+      cyclone.io/template-kind: build
+    annotations:
+      cyclone.io/description: >
+        Build Java source code with Maven
+  spec:
+    pod:
+      inputs:
+        arguments:
+        - name: image
+          value: maven:3.5.3-jdk-8-alpine
+        - name: cmd
+          value: |
+            cd /workspace/<repo-name> &&
```

Comment:
Directly git clone into `/workspace`? Then users do not need to car about `repo-name`.

Reply:
Agree, Java doesn't care about `repo-name`.

Reply:
Resolved. https://github.com/caicloud/cyclone/pull/739/files